Feature/support for vendor prefix

saberone opened the pull request:
### Requirements

### Description of the Change

I added the option to add a "vendorprefix" to the projectname in Visual Studio and the unicorn folders.

The generator now also prompts for a vendorprefix and the LayerPrefixedProjectName will contain the prefix if it's not empty.

### Benefits

### WORLD DOMINATION

But seriously, it's makes it possible to have a better organised folder structure.

### Possible Drawbacks

Users are being asked an additional question.

saberone committed:
Added support for vendor prefix

You can now enter an optional vendor prefix. This way it looks like : /Project|Feature|Foundation/[optional vendor prefix/]/Projectname.

saberone committed:
Added support for vendor prefix

You can now enter an optional vendor prefix. This way it looks like : /Project|Feature|Foundation/[optional vendor prefix/]/Projectname.

saberone committed:
Updated README.md. Now it also lists vendorprefix variable

saberone committed:
Updated README.md. Now it also lists vendorprefix variable

saberone committed:
Merged branch feature/support-for-vendor-prefix into feature/support-for-vendor-prefix

saberone committed:
Removed console logging statements.

saberone committed:
Handled empty vendorprefix correctly

Removed console logging statements. And an empty vendorprefix wasn't handled correctly. I was under the assumption that the variable would be undefined, but it's an empty string.

saberone committed:
Merged branch feature/support-for-vendor-prefix into feature/support-for-vendor-prefix

Saturate reviewed (changes requested):
Thanks for the PR :-) However I have two minor issues:

If you don't define the optional Vendor Prefix, it will create the files as following:

`src\Feature\FeatureName\code\Feature..FeatureName.csproj`

Where there is an extra `.`

Another thing is that the Vendor Prefix needs to be the first (`VendorPrefix.Layer.Name`) in the file name, like documented on http://helix.sitecore.net/principles/visual-studio/projects.html#projects

PS. If you merge the master into your branch you will have some more tests. You could write a test for your addition and it's perfect!

saberone committed:
Merge remote-tracking branch 'upstream/master' into feature/support-for-vendor-prefix

saberone committed:
Added tests for the vendorprefix functionality
The empty vendorprefix issue should be handled correctly. And the vendorprefix now actually is a prefix. ;)

Saturate reviewed (approved):
Awesome! The tests is a really nice touch, I also tested it manually. It works like one would expect.

I would say this is ready to merge. But I'll let another one review, and merge.

(PS I would suggest to the merger to do a squash to keep our git history clean. The author could also do this.)

saberone commented:
I agree with the squashing. What I find a bit strange, is that Github doesn't let the PR author write a squash suggestion as part of the PR.
